Tools/build/.warningignore_macos

```diff
@@ -3,6 +3,7 @@
 # Keep lines sorted lexicographically to help avoid merge conflicts.
 # Format example:
 # /path/to/file (number of warnings in file)
+Modules/_sqlite/clinic/connection.c.h 6
```

Comment:
I think this will be unneeded post #133605.

Reply:
This is needed to pass CI right now; #133605 should remove it after update.

Reply:
cc. @serhiy-storchaka

Reply:
The alternative is that I pull 133605 into this branch before merging it.